solr/solrj/src/java/org/apache/solr/client/solrj/impl/ConcurrentUpdateSolrClient.java

```diff
@@ -121,9 +121,11 @@ protected ConcurrentUpdateSolrClient(String solrServerUrl,
   
   protected ConcurrentUpdateSolrClient(Builder builder) {
     this.internalHttpClient = (builder.httpClient == null);
-    this.client = new HttpSolrClient.Builder(builder.baseSolrUrl)
-        .withHttpClient(builder.httpClient)
-        .build();
+    HttpSolrClient.Builder httpClientBuilder = new HttpSolrClient.Builder(builder.baseSolrUrl)
+        .withHttpClient(builder.httpClient);
+    httpClientBuilder.connectionTimeoutMillis = builder.connectionTimeoutMillis;
+    httpClientBuilder.socketTimeoutMillis = builder.socketTimeoutMillis;
```

Comment:
Maybe you should use with* methods and do the build as the previous version.

```
        .withConnectionTimeout(builder.connectionTimeoutMillis)
        .withSocketTimeout(builder.socketTimeoutMillis)
```

Reply:
Right. I've updated the pull request accordingly.